odl/operator/fspace_ops.py

```diff
@@ -0,0 +1,412 @@
+# Copyright 2014, 2015 Jonas Adler
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Default operators defined on any `FunctionSpace`."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+import numpy as np
+
+# ODL imports
+from odl.operator.operator import Operator
+from odl.set.pspace import ProductSpace
+
+
+__all__ = ('DiscreteGradient', 'DiscreteDivergence')
+
+
+def discrete_part_deriv(f, axis=0, dx=1.0, edge_order=2,
```

Comment:
This should be an operator IMO. And as such, it should not be defined on NumPy arrays immediately but rather for `DiscreteLp` vectors. If we really need a pure array version (name TBD, not `discrete_part_deriv`, rather `divided_diffs` or `finite_diff`) that one could be a helper function similar to this guy, which is then used internally by the operator.